### ISSUE TITLE
Unmet peer dependency react-dom in React Native project

### ISSUE THREAD

**Issue by betoharres** [styled-components/styled-components#2555]
Hi folks, I'm avoiding to have warnings in my project after running `yarn add`, but I'd like to use this awesome library, and after installing `styled-components` in my React Native project I noticed the unmet dependency warning about react-dom, which is not the case for RN projects. 
What you guys thinks that can be done to avoid this warning? Move to a separate repo? 

thanks all ❤️ 

## Environment

`npx envinfo --system --binaries --npmPackages styled-components,babel-plugin-styled-components --markdown --clipboard`

```
## System:
 - OS: macOS High Sierra 10.13.6
 - CPU: (4) x64 Intel(R) Core(TM) i5-2415M CPU @ 2.30GHz
 - Memory: 6.93 GB / 16.00 GB
 - Shell: 5.3 - /bin/zsh
## Binaries:
 - Node: 11.14.0 - /usr/local/bin/node
 - Yarn: 1.15.2 - /usr/local/bin/yarn
 - npm: 6.9.0 - /usr/local/bin/npm
 - Watchman: 4.9.0 - /usr/local/bin/watchman
## npmPackages:
 - styled-components: ^4.2.0 => 4.2.0
```

## Steps to reproduce
`react-native init testsc && cd testsc && yarn add styled-components`

## Expected Behavior
No warning about unmet dependency of react-dom

## Actual Behavior
`warning " > styled-components@4.2.0" has unmet peer dependency "react-dom@>= 16.3.0".`

**Comment by probablyup** [comment]
Just ignore it, as you know it's an irrelevant warning. We aren't planning on publishing a separate package at the moment.

**Comment by probablyup** [comment]
Although maybe using `optionalPeerDependencies` for this would work? Not really sure.
